### PR TITLE
sql: add STRAIGHT_JOIN and update comment syntax

### PR DESCRIPTION
**Pull request by lilin90**
@winoros @CaitinChen PTAL 
Via: https://github.com/pingcap/docs-cn/pull/663

### CODE REVIEW

File: sql/comment-syntax.md [hunk 1]
```diff
@@ -93,25 +93,9 @@ SELECT /*+ hint */ FROM ...;
...
-  
-- TIDB_INLJ(t1, t2)
-
-  ```
-  SELECT /*+ TIDB_INLJ(t1, t2) */ * from t1，t2 where t1.id = t2.id
-  ```
-  
-  The Hint optimizer uses the Index Nested Loop Join algorithm. This algorithm is faster in some scenarios and consumes less system resources, while it may be slower in some other scenarios and consumes more system resources. For the scenarios that have a small result set (less than 10,000 lines) after the filtration of `WHERE` condition, you can try to use it. The parameter in `TIDB_INLJ()` is the candidate table of the driving table (outer table) when the query plan is created. In other words, `TIDB_INLJ(t1)` only uses `t1` as the driving table to create the query plan.
-  
+For details about the optimizer that TiDB supports, see [Optimizer hint](tidb-specific.md#optimizer-hint).
```
Comment (zz-jason) [inline sql/comment-syntax.md]: s/optimizer/optimizer hints/

**Comment by CaitinChen** [comment]
LGTM

**Comment by lilin90** [comment]
@zz-jason PTAL

**Comment by zz-jason** [comment]
LGTM

**Pull request merged**